tests/common/fixtures/advanced_reboot.py

```diff
@@ -0,0 +1,510 @@
+                "reboot_type" :self.rebootType,
+                "portchannel_ports_file" : self.rebootData['portchannel_interfaces_file'],
+                "vlan_ports_file" : self.rebootData['vlan_interfaces_file'],
+                "ports_file" : self.rebootData['ports_file'],
+                "dut_mac" : self.rebootData['dut_mac'],
+                "dut_vlan_ip" : self.rebootData['dut_vlan_ip'],
+                "default_ip_range" : self.rebootData['default_ip_range'],
+                "vlan_ip_range" : self.rebootData['vlan_ip_range'],
+                "lo_v6_prefix" : self.rebootData['lo_v6_prefix'],
+                "arista_vms" : self.rebootData['arista_vms'],
+                "preboot_files" : self.prebootFiles,
+                "preboot_oper" : prebootOper,
+                "inboot_oper" : inbootOper,
+                "nexthop_ips" : self.rebootData['nexthop_ips'],
+                "allow_vlan_flooding" : self.allowVlanFlooding,
+                "sniff_time_incr" : self.sniffTimeIncr,
+                "setup_fdb_before_test" : True,
+                "vnet" : self.vnet,
+                "vnet_pkts" : self.vnetPkts,
+            },
+            log_file=u'/tmp/advanced-reboot.ReloadTest.log'
+        )
+
+    def __restorePrevImage(self):
+        '''
+        Resotre previous image and reboot DUT
+        '''
+        logger.info('Restore current image')
+        self.duthost.shell('sonic_installer set_default {0}'.format(self.currentImage))
+
+        rebootDut(
+            self.duthost,
+            self.localhost,
+            reboot_type=self.rebootType.replace('-reboot', ''),
+            wait = 180 + self.readyTimeout
+        )
+
+    def tearDown(self):
+        '''
+        Tears down test case. It also verifies that config_db.json exists.
+        '''
+        logger.info('Running test tear down')
+        if 'warm-reboot' in self.rebootType and self.newSonicImage is not None:
+            logger.info('Save configuration after warm rebooting into new image')
+            self.duthost.shell('config save -y')
+
+        result = self.duthost.shell('stat /etc/sonic/config_db.json')
+        assert len(result['stderr_lines']) == 0, '/etc/sonic/config_db.json is missing'
+
+        self.__runScript(['remove_ip.sh'], self.ptfhost)
+
+        if not self.stayInTargetImage:
+            self.__restorePrevImage()
+
+@pytest.fixture
+def get_advanced_reboot(request, testbed_devices, testbed):
+    '''
+    Pytest test fixture that provides access to AdvancedReboot test fixture
+        @param request: pytest request object
+        @param testbed_devices: fixture provides information about testbed devices
+        @param testbed: fixture provides information about testbed
+    '''
+    instances = []
+
+    def get_advanced_reboot(**kwargs):
+        '''
+        API that returns instances of AdvancedReboot class
+        '''
+        assert len(instances) == 0, "Only one instance of reboot data is allowed"
```

Comment:
Since we only allow 1 instance, do we need this to be a list?

Reply:
I tried to get away without it, however looks like list as an object has visibility to nested method, other simple object did not.

Reply:
That's really strange. What operations were you trying to do with the object?

Reply:
for whatever reason python treat object differently. If an instance of class is defined, it considers it as immutable and it doesn't bind it to local var. Where a list binds OK!

Reply:
Yup you are totally right, I updated my review comment. 🤦‍♂